tools/pylib/_boutcore_build/boutcore.pyx.jinja

```diff
@@ -400,29 +431,34 @@ cdef class {{ field.field_type }}:
         data : array_like
             Values to be set. Must be {{ field.ndims }}D and match the size of the slicing object.
         """
-        if len(slices)!={{ field.ndims }}:
-            raise IndexError("This is a {{ field.ndims }}D object!")
-        dims = []
-        import numbers
-        inds = _resolve_slices(slices,
-                              [{{ field.makelist("self.cobj.getN$d()") }}])
-        if isinstance(data_,{{ field.field_type }}):
-            # get all to get a numpy array
-            data_ = data_[{{ field.makelist(":") }}]
-        elif isinstance(data_,numbers.Number):
-            c_set_{{ field.fdd }}_part_(self.cobj,<double>data_,{{ field.makelist("inds[$i][0],inds[$i][1],inds[$i][2]") }})
+        cdef np.ndarray[double, mode="c", ndim=1] data_flat
+        slices = self._sanitise_slices(slices)
+        if isinstance(slices, tuple) and all([isinstance(x, slice) for x in slices]):
+            dims = []
+            import numbers
```

Comment:
This should probably be at the top of the file, rather than in a method

Reply:
I noticed there are a few more imports within methods:
`boutdata.collect`, `traceback`, `sys`, `gc`.

I would like to keep `boutdata` as a late import, as I think it is rather slow to import (250 ms on warm cache), and not often needed. `traceback` also seems to add ~10 ms, while the other ones seem to be negligible. Currently `import boutcore` takes around 160 ms, so `boutdata` would more than double that ...

(tested with python 3.11.0rc2, with `time python -c "import <x>"`)